examples/device/net_lwip_webserver/src/usb_descriptors.c

```diff
@@ -155,15 +155,16 @@ static uint16_t _desc_str[32];
 // Application return pointer to descriptor, whose contents must exist long enough for transfer to complete
 uint16_t const* tud_descriptor_string_cb(uint8_t index, uint16_t langid)
 {
-  (void)langid;
+  (void) langid;
 
-  unsigned chr_count = 0;
```

Comment:
I'm not a fan of uint8_t for counter variables, since it may require extra instructions to truncate the register values to 8-bit (since the result of all arithmetic is 32-bit). But, in this case, this change doesn't make any difference in the optimized firmware image (since the compiler already knows that the variable is limited to small numbers, I think it'll use 32-bit, anyway).

I vote to leave this as unsigned, but no functional issue with either choice (since they compile to the same thing).

Reply:
Amen.  I'm also not a fan of uint8_t for all the same reasons.

Reply:
I do prefer the fixed size integer since we don't know which MCU are running. Yesterday I attempted to compile lwip_webserver with MSP430 (16-bit), almost get it running. But lwip throw out too many integer overflow warning e.g `1 << 24` so I have to skip it. Though since it is  uint8_t, it make no difference, we can keep it as it is if you like it that way, though I would suggested to put it `unsigned int` as I like to be explicit although it is the same.

Reply:
Hmm. maybe `unsigned int`, then. It'll be 16-bit on MSP430, and 32-bit elsewhere, which should be fine.